### PR TITLE
[FIX] Make sure game is available before auto updating

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**
Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/main.ts [hunk 1]
```diff
@@ -629,7 +630,8 @@ ipcMain.handle('runWineCommand', async (e, args) => runWineCommand(args))
 ipcMain.handle('checkGameUpdates', async (): Promise<string[]> => {
   let oldGames: string[] = []
   const { autoUpdateGames } = GlobalConfig.get().getSettings()
-  for (const runner in libraryManagerMap) {
+  let runner: Runner = 'sideload'
+  for (runner in libraryManagerMap) {
```
Comment (CommandMC) [inline src/backend/main.ts:634]: Was this change necessary?

File: src/backend/storeManagers/index.ts [hunk 2]
```diff
@@ -12,7 +12,8 @@ import { GameManager, LibraryManager } from 'common/types/game_manager'
 import { logInfo, RunnerToLogPrefixMap } from 'backend/logger/logger'
 
 import { addToQueue } from 'backend/downloadmanager/downloadqueue'
-import { DMQueueElement, GameInfo } from 'common/types'
+import { DMQueueElement, GameInfo, Runner } from 'common/types'
+import { isGameAvailable } from 'backend/api/helpers'
 interface GameManagerMap {
   [key: string]: GameManager
 }
```
Comment (CommandMC) [inline src/backend/storeManagers/index.ts:19]:
```suggestion
type GameManagerMap = {
  [key in Runner]: GameManager
}
```

File: src/backend/main.ts (hunk 1 above)
Comment (Etaash-mathamsetty) [reply src/backend/main.ts:634]: yes you can't cast a string to a runner, but you can cast a runner to a string

Comment (Etaash-mathamsetty) [reply src/backend/main.ts:634]: nvm the cast worked this time idk

**Review by flavioislima — APPROVED** [review]
Good catch, thanks for fixing that!
